sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationEditFragment.java

```diff
@@ -177,33 +192,66 @@ public void onAfterLayoutBinding(final FragmentImmunizationEditLayoutBinding con
 		contentBinding.immunizationValidUntil.initializeDateField(getFragmentManager());
 		contentBinding.immunizationLastInfectionDate.initializeDateField(getFragmentManager());
 
-		contentBinding.immunizationMeansOfImmunization.addValueChangedListener(e -> {
-			if (e.getValue() == MeansOfImmunization.OTHER || e.getValue() == MeansOfImmunization.RECOVERY) {
-				contentBinding.immunizationImmunizationManagementStatus.setValue(ImmunizationManagementStatus.COMPLETED);
-				contentBinding.immunizationImmunizationManagementStatus.setEnabled(false);
-			}
-			if (e.getValue() == MeansOfImmunization.VACCINATION || e.getValue() == MeansOfImmunization.VACCINATION_RECOVERY) {
-				contentBinding.immunizationVaccinationLayout.setVisibility(View.VISIBLE);
-				contentBinding.immunizationNumberOfDoses.setEnabled(true);
-			} else {
-				contentBinding.immunizationVaccinationLayout.setVisibility(View.GONE);
-			}
-			if (e.getValue() == MeansOfImmunization.RECOVERY || e.getValue() == MeansOfImmunization.VACCINATION_RECOVERY) {
-				contentBinding.immunizationRecoveryLayout.setVisibility(View.VISIBLE);
-				contentBinding.immunizationRecoveryDate.setEnabled(true);
-				contentBinding.immunizationPositiveTestResultDate.setEnabled(true);
-
-				if (record.getRelatedCase() != null) {
-					contentBinding.linkCase.setVisibility(View.GONE);
-				} else {
-					contentBinding.openLinkedCase.setVisibility(View.GONE);
+
+		contentBinding.immunizationMeansOfImmunization.addValueChangedListener(new ValueChangeListener() {
+
+			private MeansOfImmunization currentMeansOfImm = record.getMeansOfImmunization();
+
+			@Override
+			public void onChange(ControlPropertyField e) {
+				MeansOfImmunization meansOfImmunization = (MeansOfImmunization) e.getValue();
+
+				if (currentMeansOfImm != meansOfImmunization) {
+					if (meansOfImmunization == MeansOfImmunization.OTHER || meansOfImmunization == MeansOfImmunization.RECOVERY) {
+						contentBinding.immunizationImmunizationManagementStatus.setValue(ImmunizationManagementStatus.COMPLETED);
+						contentBinding.immunizationImmunizationManagementStatus.setEnabled(false);
+
+						removeVaccinationsConfirmation();
+						if (meansOfImmunization == MeansOfImmunization.VACCINATION || meansOfImmunization == MeansOfImmunization.VACCINATION_RECOVERY) {
```

Comment:
There's something off with this code. This if condition is always false because it's nested in the if clause in line 205.

Reply:
good catch, not sure how i managed this one
corrected it